Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/FileTransporter.cs

```diff
@@ -4,459 +4,446 @@
-        private readonly string[] _filePaths;
-        private RenamingPolicy _renamingPolicy;
+    private readonly Guid _instanceId;
+    private readonly ISFTPLogger _logger;
+    private readonly BatchContext _batchContext;
+    private readonly string[] _filePaths;
+    private readonly RenamingPolicy _renamingPolicy;
 
-        internal FileTransporter(ISFTPLogger logger, BatchContext context, Guid instanceId)
-        {
-            _logger = logger;
-            _batchContext = context;
-            _instanceId = instanceId;
-            _renamingPolicy = new RenamingPolicy(_batchContext.Info.TransferName, _instanceId);
+    internal FileTransporter(ISFTPLogger logger, BatchContext context, Guid instanceId)
+    {
+        _logger = logger;
+        _batchContext = context;
+        _instanceId = instanceId;
+        _renamingPolicy = new RenamingPolicy(_batchContext.Info.TransferName, _instanceId);
 
-            _result = new List<SingleFileTransferResult>();
-            _filePaths = ConvertObjectToStringArray(context.Source.FilePaths);
+        _result = new List<SingleFileTransferResult>();
+        _filePaths = ConvertObjectToStringArray(context.Source.FilePaths);
 
-            SourceDirectoryWithMacrosExtended = _renamingPolicy.ExpandDirectoryForMacros(context.Source.Directory);
-            DestinationDirectoryWithMacrosExtended = _renamingPolicy.ExpandDirectoryForMacros(context.Destination.Directory);
-        }
+        SourceDirectoryWithMacrosExtended = _renamingPolicy.ExpandDirectoryForMacros(context.Source.Directory);
+        DestinationDirectoryWithMacrosExtended = _renamingPolicy.ExpandDirectoryForMacros(context.Destination.Directory);
+    }
 
-        private List<SingleFileTransferResult> _result { get; set; }
+    private List<SingleFileTransferResult> _result { get; set; }
 
-        private string SourceDirectoryWithMacrosExtended { get; set; }
+    private string SourceDirectoryWithMacrosExtended { get; set; }
 
-        private string DestinationDirectoryWithMacrosExtended { get; set; }
+    private string DestinationDirectoryWithMacrosExtended { get; set; }
 
-        /// <summary>
-        /// Executes file transfers.
-        /// </summary>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
-        /// <exception cref="DirectoryNotFoundException"></exception>
-        /// <exception cref="FileNotFoundException"></exception>
-        /// <exception cref="ArgumentException"></exception>
-        public FileTransferResult Run(CancellationToken cancellationToken)
+    /// <summary>
+    /// Executes file transfers.
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <exception cref="DirectoryNotFoundException"></exception>
+    /// <exception cref="FileNotFoundException"></exception>
+    /// <exception cref="ArgumentException"></exception>
+    public FileTransferResult Run(CancellationToken cancellationToken)
```

Comment:
You are not using cancellation token, so it can be removed, or use it somewhere in the method.

Reply:
Added check for cancellationToken.ThrowIfCancellationRequested() to the inside foreach loop.